From 614e86f85c0e3d5b85008d618dd0c13d46037822 Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Tue, 19 Sep 2023 14:09:21 +0530
Subject: [PATCH] camera-ext/ivi: Add camera service and config file

Added the required packages, configuration file for camera HAL
used in IVI.

Tracked-On: OAM-112309
Signed-off-by: shivasku82 <shiva.kumara.rudrappa@intel.com>
---
 groups/camera-ext/ivi/files.spec            |  1 +
 groups/camera-ext/ivi/ivi_camera_config.xml | 50 +++++++++++++++++++++
 groups/camera-ext/ivi/product.mk            |  7 +--
 3 files changed, 55 insertions(+), 3 deletions(-)
 create mode 100644 groups/camera-ext/ivi/ivi_camera_config.xml

diff --git a/groups/camera-ext/ivi/files.spec b/groups/camera-ext/ivi/files.spec
index 80576eb..5b402a0 100644
--- a/groups/camera-ext/ivi/files.spec
+++ b/groups/camera-ext/ivi/files.spec
@@ -1,2 +1,3 @@
 [extrafiles]
 external_camera_config.xml: "external camera parameters"
+ivi_camera_config.xml: "ivihal camera parameters"
diff --git a/groups/camera-ext/ivi/ivi_camera_config.xml b/groups/camera-ext/ivi/ivi_camera_config.xml
new file mode 100644
index 0000000..d761753
--- /dev/null
+++ b/groups/camera-ext/ivi/ivi_camera_config.xml
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<IVICamera>
+    <Provider>
+        <CameraIdOffset>0</CameraIdOffset>
+        <ignore> <!-- Internal video devices to be ignored by external camera HAL -->
+        </ignore>
+    </Provider>
+    <!-- See ExternalCameraUtils.cpp for default values of Device configurations below
+-->
+    <Device>
+        <Path>/dev/video0</Path>
+        <Driver>uvcvideo</Driver>
+        <NumVideoBuffers count="2"/>
+        <NumStillBuffers count="2"/>
+        <Setting width="1920" height="1080" fps="30" format="UYVY"/>
+        <Bind>
+            <VirtId>10</VirtId>
+        </Bind>
+    </Device>
+    <Device>
+        <Path>/dev/video1</Path>
+        <Driver>intel-ipu6-isys</Driver>
+        <NumVideoBuffers count="2"/>
+        <NumStillBuffers count="2"/>
+        <Setting width="1920" height="1080" fps="30" format="UYVY"/>
+        <Bind>
+            <VirtId>11</VirtId>
+            <VirtId>12</VirtId>
+        </Bind>
+    </Device>
+
+    <Device>
+        <Path>/dev/video5</Path>
+        <Driver>uvcvideo</Driver>
+        <NumVideoBuffers count="2"/>
+        <NumStillBuffers count="2"/>
+        <Setting width="1920" height="1080" fps="30" format="YUYV"/>
+        <Bind>
+            <VirtId>13</VirtId>
+        </Bind>
+    </Device>
+
+    <Loopback>
+        <PairId>50</PairId>
+        <SrcCamera>/dev/video0</SrcCamera>
+        <Driver>v4l2 loopback</Driver>
+        <Setting width="1920" height="1080" fps="30" format="UYVY"/>
+    </Loopback>
+    
+</IVICamera>
diff --git a/groups/camera-ext/ivi/product.mk b/groups/camera-ext/ivi/product.mk
index 2beb836..e9ad382 100644
--- a/groups/camera-ext/ivi/product.mk
+++ b/groups/camera-ext/ivi/product.mk
@@ -2,12 +2,13 @@
 PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.camera.autofocus.xml:vendor/etc/permissions/android.hardware.camera.autofocus.xml \
     frameworks/native/data/etc/android.hardware.camera.external.xml:vendor/etc/permissions/android.hardware.camera.external.xml \
-    $(LOCAL_PATH)/{{_extra_dir}}/external_camera_config.xml:vendor/etc/external_camera_config.xml
+    $(LOCAL_PATH)/{{_extra_dir}}/external_camera_config.xml:vendor/etc/external_camera_config.xml \
+    $(LOCAL_PATH)/{{_extra_dir}}/ivi_camera_config.xml:vendor/etc/ivi_camera_config.xml
 
 
 # External camera service
-PRODUCT_PACKAGES += android.hardware.camera.provider@2.4-external-service \
-                    android.hardware.camera.provider@2.4-impl
+PRODUCT_PACKAGES += android.vendor.hardware.camera.provider@2.4-ivi-service \
+                    android.hardware.camera.provider@2.4-impl-intel
 
 
 PRODUCT_PACKAGES += MultiCameraApp
-- 
2.17.1

